filebeat/input/log/log.go

```diff
@@ -58,8 +60,10 @@ func (f *Log) Read(buf []byte) (int, error) {
 
 		n, err := f.fs.Read(buf)
 		if n > 0 {
+			f.closeMutex.Lock()
```

Comment:
This is to prevent a race for f.offset and f.lastTimeRead. It's probably better to introduce getters which have the lock as otherwise this causes issue every time the variable is used somewhere else.